aws_lambda_powertools/utilities/parameters/secrets.py

```diff
@@ -126,11 +130,159 @@ def _get(self, name: str, **sdk_options) -> str | bytes:
+                sdk_options.pop("NextToken")  # pragma: no cover
+
+            try:
+                response = self.client.batch_get_secret_value(**sdk_options)
+            except Exception as exc:
+                raise GetSecretError(f"Failed to retrieve secrets: {str(exc)}") from exc
+
+            # Process successful secrets
+            for secret in response.get("SecretValues", []):
+                secret_name = secret["Name"]
+
+                # Extract secret value (SecretString or SecretBinary)
+                if "SecretString" in secret:
+                    secrets[secret_name] = secret["SecretString"]
+                elif "SecretBinary" in secret:
+                    secrets[secret_name] = secret["SecretBinary"]
+
+            # Check if there are more results
+            next_token = response.get("NextToken")
+            if not next_token:
+                break
+
+        # If no secrets were found, raise an error
+        if not secrets:
+            raise GetSecretError(f"No secrets found matching the provided names: {names}")
+
+        return secrets
+
+    def get_multiple(  # type: ignore[override]
+        self,
+        names: list[str],
+        max_age: int | None = None,
+        transform: TransformOptions = None,
+        raise_on_transform_error: bool = False,
+        force_fetch: bool = False,
+        **sdk_options,
+    ) -> dict[str, Any]:
+        """
+        Retrieve multiple secrets by name from AWS Secrets Manager
+
+        Parameters
+        ----------
+        names: list[str]
+            List of secret names to retrieve
+        max_age: int, optional
+            Maximum age of the cached value
+        transform: str, optional
+            Optional transformation of the parameter value. Supported values
+            are "json" for JSON strings and "binary" for base 64 encoded values.
+        raise_on_transform_error: bool, optional
+            Raises an exception if any transform fails, otherwise this will
+            return a None value for each transform that failed
+        force_fetch: bool, optional
+            Force update even before a cached item has expired, defaults to False
+        sdk_options: dict, optional
+            Arguments that will be passed directly to the underlying API call
+
+        Returns
+        -------
+        dict[str, str | bytes | dict]
+            Dictionary mapping secret names to their values
+
+        Raises
+        ------
+        GetParameterError
+            When the parameter provider fails to retrieve secrets
+        TransformParameterError
+            When the parameter provider fails to transform a secret value
+        """
+        if not names:
+            raise GetSecretError("You must provide at least one secret name")
+
+        # Create a unique cache key for this batch of secrets
+        # Use sorted names to ensure consistent caching regardless of order
+        cache_key_name = "|".join(sorted(names))
```

Comment:
I'm wondering if it would make sense to cache the secrets individually rather than as a single group. Especially when working in more complex code bases where the secrets might be retrieved from different modules, having a more granular cache could be useful.

For example, assuming I'm retrieving `paramaters.get_secrets_by_name(names=["a", "b"])` instead of caching them together, each one would have its own cache key & entry. Thanks to this, if one is retrieved separately before/after it would benefit from caching instead of going to Secrets Manager multiple times.

With that said, I don't know if it's worth the effort at this stage - if it's too complex we can probably postpone it until there's demand.

Reply:
I still think we can wait for customer demand. I can't assume the region will always be present in the `Filters` field, as the customer can use different keys such as tags, all, and others, and I always need to pass sdk_options/Filter as part of the key, and in this case the caching mechanism won't affect subsequent calls using `get_secret`.

Let's release it as is and see if clients actually need it.

Reply:
Can you please review again?